src/exo/frontend/pyparser.py

```diff
@@ -15,6 +15,9 @@
 from ..core.prelude import *
 from ..core.extern import Extern
 
+from typing import Any, Callable, Union, NoReturn, Optional
+import copy
+from dataclasses import dataclass
```

Comment:
What are those imports? Are they necessary? If so, you should add them as dependencies to `requirements.txt`

Reply:
The imports are parts of the python standard library just to give some type annotations and reduce some clutter